### ISSUE TITLE
read_parquet - out of bounds timestamp

### ISSUE THREAD

**Issue by jarretg** [awslabs/aws-data-wrangler#296]
I am trying to read parquet files created via AWS DMS, sourced from SQL Server. Im running into issues on some files with the below error. In one file, I found a date of “3018-03-07 00:00:00.000”. This was a typo by the user who entered the data, but now I can’t process the file. Is there any way to ignore these bad dates and still read in the file? Or any other workarounds?

_ArrowInvalid: Casting from timestamp[us] to timestamp[ns] would result in out of bounds timestamp:_

I am using aws wrangler version 1.5.0.

**Comment by igorborgest** [comment]
Hi @jarretg, thanks for reach out.

Yep, for sure this year is out of range for timestamps with nanoseconds resolution. And Wrangler + PyArrow use this resolution by default behind the scenes.

To allow you to force the load, I've just added the argument `safe` on `wr.s3.read_parquet()`. By default it will be `True`, but you can pass `False` to force the corruption of the timestamp value out of range, but the rest will be intact.

Example:
```py3
wr.s3.read_parquet("YOUR_PATH", safe=False)
````

It is still on our development branch. Could you test it, and give us some feedback?

To install the development branch:

```
pip install git+https://github.com/awslabs/aws-data-wrangler.git@dev
```

If everything work fine I will release it on version `1.6.0` in the next hours.

**Comment by jarretg** [comment]
Hello Igor!  Thank you for the extremely fast turnaround!  I did test the dev branch and it does now successfully read the file, you get my vote to push it into 1.6.0.

The only caveat is that the dates that are read are changed, here are some examples:

- source = 4200-01-12 00:00:00.000, wrangler = 1861-10-25 01:41:45.161793536
- source = 9999-12-30 00:00:00.000, wrangler = 1816-03-28 05:56:08.066277376
- source = 9999-12-31 00:00:00.000, wrangler = 1816-03-29 05:56:08.066277376
- source = 4200-01-20 00:00:00.000, wrangler = 1861-11-02 01:41:45.161793536
- source = 8200-03-05 00:00:00.000, wrangler = 1770-01-29 04:39:49.194932224

It seems as if this will be resolved in the future release of pyarrow 1.0.0.  https://issues.apache.org/jira/browse/ARROW-5359

**Comment by igorborgest** [comment]
@jarretg you are right, wrangler will mess in the out of range values.

We will stay tuned on this PyArrow ticket, thanks!

**Comment by jarretg** [comment]
Will you post on here when this has been pushed into 1.6.0?

When it goes to 1.6.0, will I be able to get the lambda layer and the Glue .whl file with the change as well?

**Comment by igorborgest** [comment]
Released on version [1.6.0](https://github.com/awslabs/aws-data-wrangler/releases/tag/1.6.0)! Lambda and Glue artefacts available.

**Comment by igorborgest** [comment]
Hi @jarretg !

It is totally off-topic, but we are stating a [_"Who uses AWS Data Wrangler?"_](https://github.com/awslabs/aws-data-wrangler/tree/dev#who-uses-aws-data-wrangler) section. So feel free to add yourself if you want 😄 .
